### DISCUSSION TITLE
Deprecated add-path

### DISCUSSION THREAD

**Discussion by w3sip** [community/community#25481] [Actions]
<p>I’m trying to replace the deprecated add-path. However, while the first fragment works, the new one does not (checkout action doesn’t find git executable). What am I doing wrong here?</p>
<p>==== &gt; original</p>
<pre><code class="lang-auto">- name: Setup environment (Windows)
       if: runner.os == 'Windows'
       shell: powershell
       run: |
            echo "::add-path::c:\git\bin"
            echo "::add-path::c:\git\cmd"

- uses: actions/checkout@v2
  with:
        lfs: true
        submodules: false
        fetch-depth: 0

</code></pre>
<p>==== &gt; new</p>
<pre><code class="lang-auto">     - name: Setup environment (Windows)
       if: runner.os == 'Windows'
       run: |
              echo "c:\git\bin" &gt;&gt; $GITHUB_PATH
              echo "c:\git\cmd" &gt;&gt; $GITHUB_PATH

     - uses: actions/checkout@v2
       with:
          lfs: true
          submodules: false
          fetch-depth: 0
</code></pre>

**Comment by thboop** [comment]
<p>Hey <a class="mention" href="/w3sip">@w3sip</a>,</p>
<p>Can you please try</p>
<pre><code class="lang-auto">run: |
  echo "c:\git\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
  echo "c:\git\cmd" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
</code></pre>
<p>You may also choose to checkout our <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files" rel="noopener nofollow ugc">docs</a> which list how to do this in powershell</p>
<p>There was a similar topic recently which may be helpful, see <a href="https://github.community/t/migration-from-deprecated-add-path-on-windows/136265/4" class="inline-onebox">Migration from deprecated add-path on windows</a></p>

**Comment by w3sip** [comment]
<p>That does not seem to work.</p>
<p>EDIT: got it going with<br>
<code>echo "c:\cmake-3.18\bin" | Out-File -Append -FilePath $env:GITHUB_PATH -Encoding utf8</code></p>
<p>Two suggestions:</p>
<ol>
<li>In both forum posts, the solution doesn’t include <code>-Append</code>. At the same time, Unix shell examples and docs are shown with <code>&gt;&gt;</code>. For those of us not familiar with Powershell, making a note of that may be useful.</li>
<li>The docs suggest the Powershell way only in context of UTF8 encoding. In fact, this form seems to be the only way to get things going on Windows. May want to make a note of that, too.</li>
</ol>

**Comment by thboop** [comment, marked as answer]
<aside class="quote group-github-employees" data-username="thboop" data-post="2" data-topic="136621">
<div class="title">
<div class="quote-controls"></div>
 thboop:</div>
<blockquote>
<pre><code class="lang-auto">run: |
  echo "c:\git\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
  echo "c:\git\cmd" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
</code></pre>
</blockquote>
</aside>
<p>Hey <a class="mention" href="/w3sip">@w3sip</a>, I apologize I forgot to add the <code>-Append</code> flag, if you add that it should work.</p>
<p>Take a look at the following workflow</p>
<pre><code class="lang-auto">name: CI
on:
  push
jobs:
  test:
    runs-on: windows-latest 
    steps:
     - run: |
         echo "c:\git\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
         echo "c:\git\cmd" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
     - run: $env:PATH
</code></pre>
<p>You can see it gets added to the path.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/7/278a322316fb89c93b805402d7dabba90b900d5c.png" data-download-href="/uploads/short-url/5DMHGqC5dWIPitHli5QPlN7KCWo.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/52323235/181091334-2a29ba85-ce87-4d0a-b0b2-e7ce3f5a6bfe.png" alt="image" data-base62-sha1="5DMHGqC5dWIPitHli5QPlN7KCWo" width="690" height="229" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/7/278a322316fb89c93b805402d7dabba90b900d5c_2_690x229.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/7/278a322316fb89c93b805402d7dabba90b900d5c_2_1035x343.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/7/278a322316fb89c93b805402d7dabba90b900d5c_2_1380x458.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/7/278a322316fb89c93b805402d7dabba90b900d5c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1404×466 61.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’ll get our docs updated so others don’t run into this issue!</p>

**Comment by w3sip** [comment]
<p>Ha, you beat me to it! Thanks!</p>
